azure-iot-device/azure/iot/device/iothub/inbox_manager.py

```diff
@@ -25,11 +25,24 @@ def __init__(self, inbox_type):
         :param inbox_type: An Inbox class that the manager will use to create Inboxes.
         """
         self._create_inbox = inbox_type
+        self.unified_message_inbox = self._create_inbox()
+        self.generic_method_request_inbox = self._create_inbox()
+        self.twin_patch_inbox = self._create_inbox()
+
+        # These inboxes are used only for non-unified receives, using APIs which are now
+        # deprecated on the client. However we need to keep them functional for backwards
+        # compatibility
         self.c2d_message_inbox = self._create_inbox()
         self.input_message_inboxes = {}
-        self.generic_method_request_inbox = self._create_inbox()
         self.named_method_request_inboxes = {}
-        self.twin_patch_inbox = self._create_inbox()
+
+        # Set this value to True if want to only use unified message mode
+        self.use_unified_msg_mode = False
```

Comment:
since use_unified_msg_mode is True if and only if self.unified_message_inbox is set, can you remove this variable and just check that unified_msg_inbox is truthy instead?  Otherwise there's a risk that use_unified_msg_mode will be True, but unified_msg_inbox will be None (or vise-versa).

Reply:
Shoot, you always allocate unified_msg_inbox even if you don't use it.  Maybe this isn't so easy. 

---
In reply to: [454407805](https://github.com/Azure/azure-iot-sdk-python/pull/620#discussion_r454407805) [](ancestors = 454407805)

Reply:
Yeah, and you can't choose which it is at instantiation, because the mode is chosen by user action after the client is created. It's a very unfortunate situation.

---
In reply to: [454410693](https://github.com/Azure/azure-iot-sdk-python/pull/620#discussion_r454410693) [](ancestors = 454410693,454407805)